### ISSUE TITLE
numpy version requirement?

### ISSUE THREAD

**Issue by dnelson86** [cbyrohl/scida#44]
```python
In [1]: from scida import load
Warning! Using default configuration. Please adjust/replace in '/u/dnelson/.config/scida/config.yaml'.
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 1
----> 1 from scida import load

File ~/.local/envs/py3/lib/python3.9/site-packages/scida/__init__.py:7
      4 # need to import interfaces to register them in registry for loading
      5 # TODO: automatize implicitly
      6 from scida.convenience import load
----> 7 from scida.customs.arepo.dataset import ArepoSnapshot
      8 from scida.customs.arepo.series import ArepoSimulation
      9 from scida.interfaces.gadgetstyle import GadgetStyleSnapshot

File ~/.local/envs/py3/lib/python3.9/site-packages/scida/customs/arepo/dataset.py:12
     10 from dask import delayed
     11 from numba import jit, njit
---> 12 from numpy._typing import NDArray
     14 from scida.discovertypes import _determine_mixins
     15 from scida.fields import FieldContainer

ModuleNotFoundError: No module named 'numpy._typing'
> /u/dnelson/.local/envs/py3/lib/python3.9/site-packages/scida/customs/arepo/dataset.py(12)<module>()
     10 from dask import delayed
     11 from numba import jit, njit
---> 12 from numpy._typing import NDArray
     13
     14 from scida.discovertypes import _determine_mixins
```

```
$ pip list

numpy                         1.21.5
```

**Comment by dnelson86** [comment]
Resolved after upgrading to numpy 1.24.4.

In this case, I also reached:

```
scida 0.2.0 requires numba<0.57,>=0.56, but you have numba 0.57.1 which is incompatible.
```

```python
In [1]: from scida import load
Warning! Using default configuration. Please adjust/replace in '/u/dnelson/.config/scida/config.yaml'.
/u/dnelson/.local/envs/py3/lib/python3.9/site-packages/scida/customs/arepo/dataset.py:643: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
```

**Comment by cbyrohl** [comment]
Regarding numba: Indeed there was some issue requiring <0.57 with other packages for now. We have to see whether this changed. Thanks for the deprecation warning, will fix.
